### ISSUE TITLE
reload/initialize Grid after dynamic HTML reload

### ISSUE THREAD

**Issue by joebordes** [nhn/tui.grid#1435]
**Summary**
My use case has a Grid loaded in an HTML div along with another bunch of fields. When one of those fields, that are external to the grid, is edited, the whole DIV is reloaded via an AJAX call. This process eliminates the grid div and reloads it. This visually works, but the data-grid-id is incremented on the new grid and then things stop working as the tui.Grid object seems to think it is still working with the previous grid.

I was looking for a call to "reset" or "initialize" tui.Grid again after the HTML load as if the tui-grid.js script had been loaded for the first time. Couldn't find anything. I tried some direct hacks to avoid the data-grid-id increment, but that didn't seem to work either. Is there a way I can do what I need to do?

**Screenshots**

![reloadtuigrid](https://user-images.githubusercontent.com/1237191/126063948-4fa68c6e-f054-4a17-a261-f65b8b4fc68f.png)

**Version**

4.17

**Comment by js87zz** [comment]
@joebordes 
Did you call `destroy` API for the old grid?
```js
grid.destroy();
// render next grid
// ...
```

**Comment by joebordes** [comment]
Yes, I did. Problem is that after destroy I need to "start" the grid again so it initializes on the new HTML object. If not I get an undefined object error.

**Comment by js87zz** [comment]
@joebordes 
If you re-render the HTML DIV rather than using API(like `resetData`) on the grid, it will not fit the grid's internal lifecycle. Because the grid doesn't know HTML DIV has been removed.
Therefore, I think that recreating the grid instance seems to be the best way.

**Comment by joebordes** [comment]
I understand. How do I recreate the grid instance?

**Comment by joebordes** [comment]
well, it definitely isn't inactive because we aren't waiting for an answer :-(

**Comment by js87zz** [comment]
@joebordes 
Sorry, As I worked on several projects at the same time, I missed checking.
In your case, when removing the `div` block, you should remove the grid through `destroy` API and recreate the instance as below. 
```js
// after reloading `div` block
grid.destroy();
grid = new Grid({
  // ...
});
```

As far as I understand the issue, there seems to be no other way to control it. Is there any other issue to consider?

**Comment by joebordes** [comment]
thanks, I'll give it a try and get back to you
